src/Microsoft.AspNet.Mvc.ModelBinding/Binders/CompositeModelBinder.cs

```diff
@@ -30,97 +31,71 @@ public CompositeModelBinder([NotNull] IEnumerable<IModelBinder> modelBinders)
+                                                            modelName: string.Empty);
+                modelBindingResult = await TryBind(newBindingContext);
             }
 
-            if (!boundSuccessfully)
+            if (modelBindingResult == null)
             {
-                return false; // something went wrong
-            }
-
-            // Only perform validation at the root of the object graph. ValidationNode will recursively walk the graph.
-            // Ignore ComplexModelDto since it essentially wraps the primary object.
-            if (newBindingContext.IsModelSet && IsBindingAtRootOfObjectGraph(newBindingContext))
-            {
-                // run validation and return the model
-                // If we fell back to an empty prefix above and are dealing with simple types,
-                // propagate the non-blank model name through for user clarity in validation errors.
-                // Complex types will reveal their individual properties as model names and do not require this.
-                if (!newBindingContext.ModelMetadata.IsComplexType &&
-                    string.IsNullOrEmpty(newBindingContext.ModelName))
-                {
-                    newBindingContext.ValidationNode = new ModelValidationNode(newBindingContext.ModelMetadata,
-                                                                               bindingContext.ModelName);
-                }
-
-                var validationContext = new ModelValidationContext(
-                    bindingContext.OperationBindingContext.MetadataProvider,
-                    bindingContext.OperationBindingContext.ValidatorProvider,
-                    bindingContext.ModelState,
-                    bindingContext.ModelMetadata,
-                    containerMetadata: null);
-
-                newBindingContext.ValidationNode.Validate(validationContext, parentNode: null);
+                return null; // something went wrong
             }
 
             bindingContext.OperationBindingContext.BodyBindingState =
                 newBindingContext.OperationBindingContext.BodyBindingState;
 
-            if (newBindingContext.IsModelSet)
+            var modelKey = bindingContext.ModelName;
+            if (modelBindingResult.IsModelSet)
             {
-                bindingContext.Model = newBindingContext.Model;
+                bindingContext.ModelMetadata.Model = modelBindingResult.Model;
+
+                // Update the model state key if we are bound using an empty prefix and it is a complex type.
+                // This is needed as validation uses the model state key to log errors. The client validation expects
+                // the errors with property names rather than the full name.
+                if (newBindingContext.ModelMetadata.IsComplexType && string.IsNullOrEmpty(modelBindingResult.Key))
+                {
+                    modelKey = modelBindingResult.Key;
+                }
```

Comment:
Ah, I see what you mean. Just make this look like whatever you think is best here. Another comment might be helpful to explain what the two cases mean, and include an **example** of what that would look like.